src/bin/commands/account-readiness/ssm.ts

```diff
@@ -1,55 +1,96 @@
   const foundInDetail = ssmParams.filter((_) => !notFoundParameters.includes(_.path));
 
-  console.log(
-    `AWS account has ${chalk.bold(`${foundInDetail.length}/${ssmParams.length}`)} SSM parameters in region ${region}`
+  const vpcs = await getVpcsInDetail(ec2Client);
+  const ssmParamsFromAccount = await getSsmParametersForVpc(ssmClient);
+  const defaultVpcs: string[] = vpcs
+    .filter((vpc) => vpc.IsDefault)
+    .map((vpc) => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- a VPC always has an ID, the type in CDK is odd!
+      return vpc.VpcId!;
+    });
+
+  const ssmParametersMatchingDefaultVpc = ssmParamsFromAccount.filter(
+    ({ Value }) => Value && defaultVpcs.includes(Value)
   );
 
-  if (foundInDetail.length > 0) {
-    console.log("\nExisting required parameters:");
+  const defaultVPCReferencePaths = ssmParametersMatchingDefaultVpc
+    .map((param) => {
+      return ssmParams.find((p) => p.path === param.Name);
+    })
+    .filter((p): p is SsmParameterPath => p !== undefined);
 
-    foundInDetail.forEach(({ path, description }) => {
-      console.log(` ✅ ${chalk.yellowBright(path)} ${chalk.dim(`(${description})`)}`);
-    });
-  }
+  return {
+    foundParameters: foundInDetail,
+    missingParameters: notFoundInDetail,
+    defaultVPCReferences: defaultVPCReferencePaths,
+  };
+};
 
-  if (notFoundInDetail.length > 0) {
-    console.log("\nMissing required parameters:");
+export const report = async (props: AwsAccountReadiness): Promise<Report> => {
+  const output = await ssmParamReadiness(props);
+  const isPass = output.missingParameters.length === 0 && output.defaultVPCReferences.length === 0;
 
-    notFoundInDetail.forEach(({ path, description }) => {
-      console.log(` ❌ ${chalk.yellowBright(path)} ${chalk.dim(`(${description})`)}`);
-    });
+  const errs = new Map<string, string>();
+  const parametersMsg =
+    "CDK uses predefined SSM parameters within patterns and constructs. For example, the EC2 APP pattern uses a parameter to determine which Kinesis stream to forward logs to. Your CDK app may fail at runtime if these parameters do not exist.";
+
+  const missingParamsMsg = "Required parameters are missing";
+  const missingParams = output.missingParameters.map((p) => `  ${chalk.yellowBright(p.path)}: ${p.description}`);
+
+  if (output.missingParameters.length > 0) {
+    errs.set(missingParamsMsg, `\n${missingParams.join("\n")}`);
   }
 
-  if (notFoundInDetail.length === 0) {
-    // no action needed
-    return 0;
-  } else {
-    console.log(
-      chalk.red(
-        `\nAWS account requires the above ${notFoundInDetail.length} SSM parameters to work with @guardian/cdk v${LibraryInfo.VERSION}`
-      )
+  if (output.defaultVPCReferences.length > 0) {
+    errs.set(
+      "Default VPC referenced in VPC parameters",
+      `it is recommended to use a custom VPC instead. Found in paths: ${output.defaultVPCReferences.join(", ")}.`
```

Comment:
Should we be more opinionated here? Rather than recommend the use of a custom VPC, require it?

Reply:
Ah, looking more closely, the rest of this change does move to requiring a custom VPC. Maybe we can change "it is recommended to" to "you should"?